packages/remix-react/components.tsx

```diff
@@ -315,28 +316,34 @@ let NavLink = React.forwardRef<HTMLAnchorElement, RemixNavLinkProps>(
 );
 NavLink.displayName = "NavLink";
 export { NavLink };
+
 /**
  * This component renders an anchor tag and is the primary way the user will
  * navigate around your website.
  *
  * @see https://remix.run/api/remix#link
  */
 let Link = React.forwardRef<HTMLAnchorElement, RemixLinkProps>(
-  ({ to, prefetch = "none", ...props }, forwardedRef) => {
-    let href = useHref(to);
+  ({ to, prefetch = "none", children, ...props }, forwardedRef) => {
+    let location = typeof to === "string" ? to : createPath(to);
+
+    let isAbsolute =
+      /^[a-z+]+:\/\//i.test(location) || location.startsWith("//");
```

Comment:
same detection we do over in router land https://github.com/remix-run/react-router/blob/dev/packages/react-router-dom/index.tsx#L421-L424